### ISSUE TITLE
[jss-extend] Extended classes with css states affecting other extended classes

### ISSUE THREAD

**Issue by noah-potter** [cssinjs/jss#436]
When I have something like:
```
    redHover: {
        '&:hover': {
            background: 'red',
        },
    },
    yellowHover: {
        extend: 'redHover',
        '&:hover': {
            background: 'yellow',
        },
    },
    supposedRedHover: {
        extend: 'redHover',
    },
```

And I put `supposedRedHover` on an element, that element will be yellow on hover.

JSS version: 6.3.0
node version: 4.5.0
npm version: 2.15.11

**Comment by kof** [comment]
Looks like a bug, extend: 'redHover' within yellowHover overwrote redHover.

**Comment by kof** [comment]
Wanna fix it?

**Comment by kof** [comment]
Just let me know when u start. The bug is in jss-extend plugin.

**Comment by noah-potter** [comment]
Submitted pr: cssinjs/jss-extend/pull/7

**Comment by kof** [comment]
merged, released version 3.0.1, thanks!
